infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -182,39 +178,30 @@ private void verifyStatusOfSnapshotSyncApply() {
                 log.info("Snapshot sync apply is complete appliedTs={}, baseTs={}", metadataResponse.getSnapshotApplied(),
                         baseSnapshotTimestamp);
                 fsm.input(new LogReplicationEvent(LogReplicationEvent.LogReplicationEventType.SNAPSHOT_APPLY_COMPLETE,
-                        new LogReplicationEventMetadata(transitionEventId, baseSnapshotTimestamp, baseSnapshotTimestamp)));
+                        new LogReplicationEventMetadata(transitionEventId, baseSnapshotTimestamp, baseSnapshotTimestamp), fsm));
+                return;
+            } else if (stopSnapshotApply.get()){
```

Comment:
Actually, this variable is not even needed. Because its set to true only on REPLICATION_STOP event, and after that we just transition to INITIALIZED state. So we don't come back to this state at all for this part to get executed.
I guess the thought for its existence was to be robust...Incase somehow even after snapshot being stopped, we come to waitSnapshotApply state. This is not possible at all. So I'm thinking it should be safe to remove it.

Please let me know if you think otherwise

Reply:
yes I agree with removing it.

However, on checking a bit further, this boolean gets unset only if entry in this state was from Initialized state.  So till now, whenever we had these transitions: Wait for Apply -> Initialized(Replication Stop) -> In Snapshot Sync -> Wait for Apply, was it remaining set to true?  It seems like a bug.